data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/sql/TypeUtils.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.runtime.operations.internal.sql;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.beans.BeanIntrospection;
+import io.micronaut.core.beans.BeanProperty;
+import io.micronaut.core.reflect.ClassUtils;
+import io.micronaut.core.reflect.ReflectionUtils;
+import io.micronaut.data.model.runtime.RuntimePersistentEntity;
+import io.micronaut.data.model.runtime.RuntimePersistentProperty;
+
+/**
+ * The type utils for DTO and persistent entity type.
+ */
+@Internal
+@SuppressWarnings({"java:S1872"})
+final class TypeUtils {
+
+    private TypeUtils() {
+    }
+
+    /**
+     * Return true if the left type is compatible or can be assigned to the right type.
+     * @param leftType The left type
+     * @param rightType The right type
+     * @return True if they are
+     */
+    static boolean areTypesCompatible(Class<?> leftType, Class<?> rightType) {
+        String rightTypeName = rightType.getName();
+        if (leftType.getName().equals(rightTypeName)) {
+            return true;
+        } else if (leftType.isInstance(rightTypeName)) {
```

Comment:
`isInstance`??

Reply:
Copied from processor but made mistake, there is no analogous method in runtime I think. Removed